pkgs/applications/misc/redshift-plasma-applet/default.nix

```diff
@@ -1,24 +1,33 @@
-{ stdenv, cmake, extra-cmake-modules, plasma-framework, kwindowsystem, redshift, fetchFromGitHub, }:
+{ stdenv, cmake, extra-cmake-modules, procps, plasma-framework, kwindowsystem, redshift, fetchgit, }:
 
 let version = "1.0.18"; in
 
 stdenv.mkDerivation {
   pname = "redshift-plasma-applet";
   inherit version;
 
-  src = fetchFromGitHub {
-    owner = "kotelnik";
-    repo = "plasma-applet-redshift-control";
-    rev = "v${version}";
-    sha256 = "122nnbafa596rxdxlfshxk45lzch8c9342bzj7kzrsjkjg0xr9pq";
+  src = fetchgit {
+      url = "https://invent.kde.org/plasma/plasma-redshift-control.git";
+      rev = "8b889f48988ede33ab28a7215e70ab2bdf1d0235";
+      sha256 = "1i8rhn4jdkhcfcia866d171j4251f9yn9a3577kli8v2wmfpcpn6";
   };
```

Comment:
changing the source but not the version is incorrect. Generally, we shouldn't use a non released version of something when there are official releases. I do see from kotelnik/** it was moved under the KDE umbrella, but still no new release. Please see the manual on `unstable` and `YYYY-MM-DD` versioning

Reply:
The most recent release is broken with the most recent version of redshift binary. <https://github.com/NixOS/nixpkgs/issues/83250>

This specific commit fixes it: <https://invent.kde.org/plasma/plasma-redshift-control/-/commit/898c3a4cfc6c317915f1e664078d8606497c4049> so I thought it would be better to update the package src.

Would it be better to directly `substituteInPlace` the fix now, and then remove the quick fix on next release?

Reply:
You can actually just patch the source instead of updating
```
patches = [
  (fetchpatch {
    url = "https://invent.kde.org/plasma/plasma-redshift-control/-/commit/898c3a4cfc6c317915f1e664078d8606497c4049.patch";
   sha256 = ...
  })
];
```

see the nixpkgs manual on fetchpatch